### ISSUE TITLE
Error on Kitty terminal when executing the new command

### ISSUE THREAD

**Issue by gustavo-hms** [mawww/kakoune#4094]
### Steps 

- compile Kakoune from master.
- run `kak` inside Kitty
- execute `new`

### Outcome 

The command fails and the `*debug*` buffer shows the following message:
```
shell stderr: <<<
/bin/sh: 2: local: not in a function
>>>
```

### Expected 

It should open a new Kitty window running `kak`.

The problem seems to be in [this line](https://github.com/mawww/kakoune/blob/da80a8cf6ae3f2993d86c2c328b099078e037c96/rc/windowing/kitty.kak#L18) and [this line](https://github.com/mawww/kakoune/blob/da80a8cf6ae3f2993d86c2c328b099078e037c96/rc/windowing/kitty.kak#L36). If I remove the `local` words, the command works.

**Comment by Screwtapello** [comment]
Yep, Shellcheck reports:

    rc/windowing/kitty.kak:18:9: warning: In POSIX sh, 'local' is undefined. [SC2039]
    rc/windowing/kitty.kak:36:9: warning: In POSIX sh, 'local' is undefined. [SC2039]

However, `local` is supported by `dash`, and `bash` in POSIX mode. Out of curiosity, what shell are you using?

**Comment by gustavo-hms** [comment]
I'm using `fish`.

**Comment by Screwtapello** [comment]
Yes, Kakoune uses `/bin/sh` to interpret shell blocks, not your interactive shell. Normally `/bin/sh` is `bash`, or a descendent of the Almquist Shell like `dash`, both of which should support `local`. Apparently your OS is unusual.

Of course, using `local` in a Kakoune script is still a bug; even though `local` is a very common extension to the POSIX standard it's still not a standard feature.

**Comment by gustavo-hms** [comment]
Hmm... this is odd, since I observed this problem in a Ubuntu machine.
